### PR TITLE
added code for TPR migration

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: service/framework.go [hunk 1]
```diff
@@ -653,3 +661,188 @@ func newPodFramework(config Config) (*framework.Framework, error) {
...
+		if err != nil {
+			logger.Log("error", fmt.Sprintf("%#v", err))
+			return
+		}
+
+		fmt.Printf("\n")
+		fmt.Printf("b start\n")
+		fmt.Printf("%s\n", b)
+		fmt.Printf("b end\n")
+		fmt.Printf("\n")
```
Comment (kopiczko) [inline service/framework.go]: I think you forgot to remove that.

File: service/framework.go [hunk 2]
```diff
@@ -653,3 +661,188 @@ func newPodFramework(config Config) (*framework.Framework, error) {
...
+		if err := json.Unmarshal(b, v); err != nil {
+			logger.Log("error", fmt.Sprintf("%#v", err))
+			return
+		}
+
+		fmt.Printf("\n")
+		fmt.Printf("v start\n")
+		fmt.Printf("%#v\n", v)
+		fmt.Printf("v end\n")
+		fmt.Printf("\n")
```
Comment (kopiczko) [inline service/framework.go]: Ditto Printf

File: service/framework.go [hunk 3]
```diff
@@ -653,3 +661,188 @@ func newPodFramework(config Config) (*framework.Framework, error) {
...
+			cro.Spec.KVM.Network.Flannel.VNI = tpo.Spec.KVM.Network.Flannel.VNI
+			cro.Spec.KVM.NodeController.Docker.Image = tpo.Spec.KVM.NodeController.Docker.Image
+			cro.Spec.KVM.Workers = toKVMWorkers(tpo.Spec.KVM.Workers)
+			cro.Spec.VersionBundle.Version = tpo.Spec.VersionBundle.Version
+
+			fmt.Printf("\n")
+			fmt.Printf("cro start\n")
+			fmt.Printf("%#v\n", cro)
+			fmt.Printf("cro end\n")
+			fmt.Printf("\n")
```
Comment (kopiczko) [inline service/framework.go]: Ditto Printf.

File: service/framework.go [hunk 4]
```diff
@@ -653,3 +661,188 @@ func newPodFramework(config Config) (*framework.Framework, error) {
...
+		}
+
+		// Create CRO in Kubernetes API.
+		{
+			// TODO enable.
+			//_, err := clientSet.ProviderV1alpha1().KVMConfigs(tpo.Namespace).Create(cro)
+			//if err != nil {
```
Comment (kopiczko) [inline service/framework.go]: I think you'll need to check IsExist.

File: service/framework.go [hunk 5]
```diff
@@ -653,3 +661,188 @@ func newPodFramework(config Config) (*framework.Framework, error) {
...
+			//	logger.Log("error", fmt.Sprintf("%#v", err))
+			//	return
+			//}
```
Comment (kopiczko) [inline service/framework.go]: Why disabled?

File: service/framework.go (hunk 1 above)
Comment (xh3b4sd) [reply service/framework.go]: I would like to keep this for the migration. I want to prepare everything and just have some PRs where I enable some commented code and then I am done.

Comment (xh3b4sd) [reply service/framework.go]: This is removed after the migration anyway.

Comment (xh3b4sd) [reply service/framework.go]: And the amount of data printed does not harm in our case.

File: service/framework.go (hunk 4 above)
Comment (xh3b4sd) [reply service/framework.go]: Good shout. Will add this.

File: service/framework.go (hunk 5 above)
Comment (xh3b4sd) [reply service/framework.go]: See my comment above.

File: service/framework.go (hunk 4 above)
Comment (xh3b4sd) [reply service/framework.go]: Done.

**Comment by xh3b4sd** [comment]
🍰

**Pull request merged**